notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/3_Forecast.ipynb

```diff
@@ -0,0 +1,972 @@
+    "                                                              \"S3Config\" : {\n",
+    "                                                                 \"Path\":s3DataPath,\n",
+    "                                                                 \"RoleArn\": role_arn\n",
+    "                                                              } \n",
+    "                                                          },\n",
+    "                                                          TimestampFormat=TIMESTAMP_FORMAT\n",
+    "                                                         )\n",
+    "    ds_import_job_arn=ds_import_job_response['DatasetImportJobArn']\n",
+    "    target_ds_import_job_arn = copy.copy(ds_import_job_arn) #used to delete the resource during cleanup\n",
+    "except Exception as e:\n",
+    "    print(e)\n",
+    "    ds_import_job_arn='arn:aws:forecast:{}:{}:dataset-import-job/{}/{}'.format(REGION_NAME, ACCOUNT_NUM, datasetArn, datasetImportJobName)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#check status -- it will change from IN PROGRESS to ACTIVE once the dataset upload is completed.\n",
+    "while True:\n",
+    "    dataImportStatus = forecast.describe_dataset_import_job(DatasetImportJobArn=ds_import_job_arn)['Status']\n",
+    "    print(dataImportStatus)\n",
+    "    if dataImportStatus != 'ACTIVE' and dataImportStatus != 'CREATE_FAILED':\n",
+    "        sleep(30)\n",
+    "    else:\n",
+    "        break\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3DataPath = 's3://{}/{}/related_time_series.csv'.format(bucket, prefix)\n",
+    "datasetImportJobName = 'forecast_DSIMPORT_JOB_RELATED'\n",
+    "try:\n",
+    "    ds_import_job_response=forecast.create_dataset_import_job(DatasetImportJobName=datasetImportJobName,\n",
+    "                                                          DatasetArn=related_datasetArn,\n",
+    "                                                          DataSource= {\n",
+    "                                                              \"S3Config\" : {\n",
+    "                                                                 \"Path\":s3DataPath,\n",
+    "                                                                 \"RoleArn\": role_arn\n",
+    "                                                              } \n",
+    "                                                          },\n",
+    "                                                          TimestampFormat=TIMESTAMP_FORMAT\n",
+    "                                                         )\n",
+    "    ds_import_job_arn=ds_import_job_response['DatasetImportJobArn']\n",
+    "    related_ds_import_job_arn = copy.copy(ds_import_job_arn) #used to delete the resource during cleanup\n",
+    "except Exception as e:\n",
```

Comment:
Catching generic "Exception" class to skip creating existing resources, but shouldn't do this as it is possible to catch different types of errors.
Use forecast.exceptions.ResourceAlreadyExistsException specifically.